### PR TITLE
UK-2337: additional logging.

### PR DESCRIPTION
**Pull request by DGrassoW**
## Context

While investigating an issue with the CoP send we weren't able to find easily the cause of an error by looking at the logs.

## Changes

Additional logs when an error occurs.

### CODE REVIEW

File: src/main/java/com/transferwise/openbanking/client/oauth/RestOAuthClient.java [hunk 1]
```diff
@@ -59,9 +59,11 @@ public AccessTokenResponse getAccessToken(GetAccessTokenRequest getAccessTokenRe
                 AccessTokenResponse.class);
             accessTokenResponse = response.getBody();
         } catch (RestClientResponseException e) {
+            log.error("Call to token endpoint failed, x-fapi-interaction-id [{}]", requestHeaders.getInteractionId(),  e);
```
Comment (DGrassoW) [inline src/main/java/com/transferwise/openbanking/client/oauth/RestOAuthClient.java:62]: @umair-akram are we happy about putting some logging here or do we want to catch/log/rethrow in the cop service?
also are we allowed to log this `x-fapi-interaction-id`? is it sensitive?

Comment (umair-akram) [reply src/main/java/com/transferwise/openbanking/client/oauth/RestOAuthClient.java:62]: With interaction-ID alone we can't identify any user related data so its safe to log

**Comment by umair-akram** [comment]
Bump the api-version as well

File: src/main/java/com/transferwise/openbanking/client/oauth/RestOAuthClient.java (hunk 1 above)
Comment (ulianaostapenko) [reply src/main/java/com/transferwise/openbanking/client/oauth/RestOAuthClient.java:62]: this is a public repo. Are you sure this is the only way how you can trace your issue?

Comment (DGrassoW) [reply src/main/java/com/transferwise/openbanking/client/oauth/RestOAuthClient.java:62]: @ulianaostapenko If I understand your question I guess I could just catch/log/re throw the exception in the cop service where it's using this client...

Comment (ulianaostapenko) [reply src/main/java/com/transferwise/openbanking/client/oauth/RestOAuthClient.java:62]: That would be awesome

Comment (umair-akram) [reply src/main/java/com/transferwise/openbanking/client/oauth/RestOAuthClient.java:62]: There is nothing wrong in logging the uuid for the request

**Pull request merged**